### ISSUE TITLE
Redirect users already logged in away from registration page

### ISSUE THREAD

**Issue by stantond** [macropin/django-registration#96]
Have I missed this functionality? Otherwise, I'd like to request it. Just a URL/Name in settings used to redirect users to somewhere else if they find their way to accounts/register/ when they're already logged in.

**Comment by dicato** [comment]
Where would you want the default redirection to go?

To achieve this functionality I use [django-braces](http://django-braces.readthedocs.org/en/latest/access.html#anonymousrequiredmixin), specifically the `AnonymousRequiredMixin` and override the `authenticated_redirect_url` in most views to point to specific other views. By default it will redirect to `settings.LOGIN_REDIRECT_URL`.

This means that my registration views usually subclass a couple different views:

```python
class RegistrationView(AnonymousRequiredMixin, DefaultRegistrationView):
    ...
```

It probably makes sense for django-registration to require django-braces and add this functionality, though I am not sure. What do others think?

**Comment by mimischi** [comment]
I'd like to have this inside the default package. If a user is already logged in, he doesn't need to view any of the login/registration themed pages.
Another thing would be the /confirm/ and /complete/ pages, which shouldn't really be displayed when accessed directly (while they don't have any functionality..)

**Comment by dicato** [comment]
@macropin do you have thoughts on leveraging django-braces to get this functionality in registration? It's how I have implemented it in a variety of cases and works well. I don't suggest re-implementing it from scratch.

**Comment by macropin** [comment]
I don't think we want to introduce a dependency for the sake of one mixin. Redirecting away to `settings.LOGIN_REDIRECT_URL` is probably a sane choice.

Are there any use cases where we *want* the old behaviour? I've certainly never encountered it.

**Comment by patroqueeet** [comment]
Dear,

what's the status here? Started missing that today :)

P

**Comment by macropin** [comment]
@patroqueeet Happy to accept a PR that redirects to `settings.LOGIN_REDIRECT_URL`

**Comment by patroqueeet** [comment]
You're still asking for another PR although my last one took more then 3 months?

**Comment by joshblum** [comment]
closed by #262 & #293
